modules/obstacleDetector/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+/******************************************************************************
+ *                                                                            *
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia (IIT)        *
+ * All Rights Reserved.                                                       *
+ *                                                                            *
+ ******************************************************************************/
+
+/**
+ * @file main.cpp
+ * @authors: Valentina Vasco <valentina.vasco@iit.it>
+ */
+
+#include <iostream>
+#include <cstdlib>
+#include <cmath>
+
+#include <yarp/os/Network.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/RFModule.h>
+#include <yarp/os/Value.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/Property.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/RpcClient.h>
+
+#include <yarp/dev/IRangefinder2D.h>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/LaserMeasurementData.h>
+
+#include <yarp/sig/Vector.h>
+#include <yarp/sig/Matrix.h>
+#include <yarp/sig/Image.h>
+
+#include <yarp/math/Math.h>
+
+#include <yarp/cv/Cv.h>
+
+#include <opencv2/core.hpp>
+
+using namespace std;
+using namespace yarp::os;
+using namespace yarp::dev;
+using namespace yarp::sig;
+using namespace yarp::math;
+using namespace cv;
+
+/****************************************************************/
+class ObstDetector : public RFModule
+{
+    double period{0.0};
+    string module_name{"obstacleDetector"};
+    string robot{"SIM_CER_ROBOT"};
```

Comment:
I would rather initialize this variable with the tag of a physical robot.

Reply:
Done.